services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +44,57 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+  if (xQueueSend(thermalMgrQueueHandle, (void *) &event, 5000/portTICK_PERIOD_MS) == pdPASS)
+  {
+    return ERR_CODE_SUCCESS;
+  }  
+  if (thermalMgrQueueHandle == NULL || event == NULL)
+  {
+    return ERR_CODE_INVALID_ARG; 
+  }
+
+  return ERR_CODE_UNKNOWN; 
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = OVER_TEMPERATURE_SHUTDOWN;
+  thermalMgrSendEvent(&interrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_t event;
+  void *const pvBuffer = &event; 
```

Comment:
you can just pass &event into xQueueReceive instead of creating this extra variable since this hurts readability in this case